frontend/play/index.js

```diff
@@ -342,6 +340,8 @@ function clearOutput() {
 // --- UI: initialization ----------------------------------------------
 
 async function initUI() {
+  initCanvas()
+  initThemeToggle("#dark-theme", "theme")
```

Comment:
These used to be done after all the other stuff in initUI, but now they are being done first before the other init functions at the end. Should these init calls all be grouped together at the end?

Reply:
nope, i have decided they are ok to be done first.
they are not async and therefore take "no" time.

Reply:
as i played everything in slow I also realized this way there is one race condition less, though I forgot which one now...